### ISSUE TITLE
evaluate my trained model 

### ISSUE THREAD

**Issue by alaa-shubbak** [open-mmlab/mmpose#2202]
thank you for your work and support. 
i would like to ask about evaluate my trained model on another evaluation metric such as pck 

i already trained my mmpose model on my custom dataset , and tested it. the evaluation was in term of AP (average precision) and AR (average recall). 

can i test the model now on, without retrained it and find the evolution in term of pck metric ? 

 i saw this [part](https://mmpose.readthedocs.io/en/latest/user_guides/configs.html?highlight=evaluation%20pck#evaluation) of config  in mmpose documentation:

`val_evaluator = dict(
    type='CocoMetric', # coco AP
    ann_file=data_root + 'annotations/person_keypoints_val2017.json') # path to annotation file
test_evaluator = val_evaluator # use val as test by default
`  

i am not sure if i update it to be pck , and test the model with such updated config , will be work or not? 

looking forward to receive a reply from your side.

**Comment by Ben-Louis** [comment]
Yeah, you are correct. In order to utilize alternative metrics, you need to modify `val_evaluator` and `test_evaluator`. For instance, if you would like to employ the PCK metric, you can reference the following example: https://github.com/open-mmlab/mmpose/blob/8dd07a75eb48eb0d482c73cf70eaa03e0a98767e/configs/hand_2d_keypoint/topdown_heatmap/onehand10k/td-hm_hrnetv2-w18_8xb64-210e_onehand10k-256x256.py#L152-L158 as example

**Comment by alaa-shubbak** [comment]
i did the update in my config file as bellow : 
<img width="471" alt="image" src="https://user-images.githubusercontent.com/10245810/231247429-e28abc0e-0779-4a88-84a7-d828e3e35210.png">

i already have the evaluation to be AP , and add the pck accuracy. i try to use the function test.py as follow
python tools/test.py configs/excavator/excavator_omin/hrnet_w32_animalpose_exc_omni_256x256_ACID.py ../../../work/shubbak/mmpose_train/results/isacc_fine_Acid_auto/hrnet/xxxx.pth --eval mAP --work-dir ...

i did not get the value of pck. what did i do wrong?

**Comment by Ben-Louis** [comment]
Apologies for the confusion, I assumed you were utilizing MMPose 1.0. However, it appears that you are using MMPose 0.x, where the available metric is tied to the dataset in use. To switch to the PCK metric, you'll need to verify if 'PCK' is included in the `allowed_metrics` within the `evaluate` method of your chosen dataset.

We highly recommend upgrading to the most recent version of MMPose, as it boasts superior design and an enhanced user experience. With the latest version, you'll have the flexibility to seamlessly switch between evaluation metrics.

**Comment by alaa-shubbak** [comment]
I would like to reopen the issue please. 
i installed the mmpose version 1.0 , and ran the command :
`python tools/test.py configs/excavator/excavator_omin/hrnet_w32_animalpose_exc_omni_256x256_ACID.py ../../../work/shubbak/mmpose_train/results/isacc_fine_Acid_2/hrnet/epoch_230.pth --eval mAP --work-dir ../../../work/shubbak/mmpose_test/result_issac_acid_fine2/hrnet_e230/`

i modified the config file to have this : 
`val_evaluator = [
    dict(type='PCKAccuracy', thr=0.2),
    dict(type='AUC'),
    dict(type='EPE'),
]
test_evaluator = val_evaluator
`

instead of this `evaluation = dict(interval=10, metric='mAP', save_best='AP')`

i got such error 
`ImportError: cannot import name 'Config' from 'mmcv' (/home/shubbak/env_pose/lib/python3.9/site-packages/mmcv/__init__.py)`

can you  please help ?

**Comment by alaa-shubbak** [comment]
I reinstall the mmpose with its github files , as the test.py has been changed in the new version. 

then i try to use it with my trained model. i got this error:

 
<img width="716" alt="image" src="https://user-images.githubusercontent.com/10245810/233217466-75e5ea74-74d6-4dae-9555-01d146a2b473.png">

i am little bit confused , is the new version support the test on old version or shall i retrain my model again from scratch? 

how can i solve this problem. please help.

**Comment by Tau-J** [comment]
Hi @alaa-shubbak , most of the code (especially the config file) in MMPose 1.0 will not be compatible with 0.x version. If possible, I suggest retraining the model based on 1.x.
